src/main/java/org/jabref/gui/mergeentries/newmergedialog/fieldsmerger/FieldMergerFactory.java

```diff
@@ -0,0 +1,31 @@
+package org.jabref.gui.mergeentries.newmergedialog.fieldsmerger;
+
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.preferences.PreferencesService;
+
+public class FieldMergerFactory {
+    private final PreferencesService preferencesService;
+
+    public FieldMergerFactory(PreferencesService preferencesService) {
+        this.preferencesService = preferencesService;
+    }
+
+    public FieldMerger create(Field field) {
+        if (field.equals(StandardField.GROUPS)) {
```

Comment:
Use a switch here?

Reply:
And also some parts use == other use equal

Reply:
I couldn't use a switch expression because `Field` is an interface. When I changed the type of the parameter to `StandardField`, it compiled fine but other parts of the codebase broke.

Reply:
Ah okay, yeah I see. Then field is the correct approach